### PR TITLE
DOCFIX: Update index() documentation with some code examples

### PR DESCRIPTION
**Pull request by scottchiefbaker**
I'm a visual learner, and examples in documentation are really helpful. I added some examples to `index()` to help other people understand it better.

### CODE REVIEW

File: pod/perlfunc.pod [hunk 1]
```diff
@@ -3585,6 +3585,20 @@ respectively.  POSITION and the return value are based at zero.
...
 
+Find characters or strings:
+
+    index("Perl is great", "P");     # Returns 0
+    index("Perl is great", "g");     # Returns 8
+    index("Perl is great", "great"); # Also returns 8
+
+Attempting to find something not there:
+
+    index("Perl is great", "Z"); # Returns -1 (not found)
```
Comment (karenetheridge) [inline pod/perlfunc.pod]: could you indent the comment to the same column as the other comments?

File: pod/perlfunc.pod [hunk 2]
```diff
@@ -3585,6 +3585,20 @@ respectively.  POSITION and the return value are based at zero.
...
+
+Using an offset to find the I<second> occurrence:
+
+    index("Perl is great", "e", 5); # Returns 10
```
Comment (karenetheridge) [inline pod/perlfunc.pod]: and here too

File: pod/perlfunc.pod (hunk 1 above)
Comment (scottchiefbaker) [reply pod/perlfunc.pod]: I just updated the PR so all the comments are on the same column

File: pod/perlfunc.pod (hunk 2 above)
Comment (scottchiefbaker) [reply pod/perlfunc.pod]: I just updated the PR so all the comments are on the same column

**Pull request merged**